source/directives/text.ts

```diff
@@ -0,0 +1,48 @@
+/// <reference path="../../node_modules/forms-js/dist/forms-js.d.ts" />
+
+module adaptor.directives {
+
+  export function TextDirective($log:ng.ILogService):ng.IDirective {
+
+    return {
+      require: '^fjsForm',
+      restrict: 'EA',
+      templateUrl: '/templates/text.html',
+
+      scope: {
+        fieldName: '@'
+      },
+
+      link: function($scope:any,
+                     $element:ng.IAugmentedJQuery,
+                     $attributes:ng.IAttributes,
+                     formForController:any):void {
+
+        if (!$scope.fieldName) {
+          $log.error('Missing required field "fieldName"');
+
+          return;
+        }
+
+        $scope.attributeMetadata = formForController.form.registerAttribute($scope.fieldName);
+        $scope.watchPath = 'attributeMetadata.form.formData.' + $scope.fieldName;
+
+        // TODO This is likely a race condition.
+        // TODO Refactor this to be shared between input directives.
+        $scope.$watch($scope.watchPath, function(value:any) {
+          $scope.bindable = formsjs.Flatten.read($scope.fieldName, $scope.attributeMetadata.form.formData);
```

Comment:
formFor has its own code for doing the same thing: [NestedObjectHelper](https://github.com/bvaughn/angular-form-for/blob/master/source/utils/nested-object-helper.ts).

I assume this particular bit of Forms JS - being that it's Angular 1x centric - would belong in the adapter rather than in the base library, but maybe we should chat about it at some point?